crates/red_knot_module_resolver/src/path.rs

```diff
@@ -0,0 +1,1179 @@
+    fn module_name_one_part_py() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo.py"), @r###"
+        ModuleName(
+            "foo",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_2_parts_dunder_init_py() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/__init__.py"), @r###"
+        ModuleName(
+            "foo",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_2_parts_dunder_init_pyi() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/__init__.pyi"), @r###"
+        ModuleName(
+            "foo",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_2_parts_no_extension() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar"), @r###"
+        ModuleName(
+            "foo.bar",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_2_parts_pyi() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar.pyi"), @r###"
+        ModuleName(
+            "foo.bar",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_2_parts_py() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar.py"), @r###"
+        ModuleName(
+            "foo.bar",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_3_parts_dunder_init_pyi() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar/__init__.pyi"), @r###"
+        ModuleName(
+            "foo.bar",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_3_parts_dunder_init_py() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar/__init__.py"), @r###"
+        ModuleName(
+            "foo.bar",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_3_parts_no_extension() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar/baz"), @r###"
+        ModuleName(
+            "foo.bar.baz",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_3_parts_pyi() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar/baz.pyi"), @r###"
+        ModuleName(
+            "foo.bar.baz",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_3_parts_py() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar/baz.py"), @r###"
+        ModuleName(
+            "foo.bar.baz",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_4_parts_dunder_init_pyi() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar/baz/__init__.pyi"), @r###"
+        ModuleName(
+            "foo.bar.baz",
+        )
+        "###);
+    }
+
+    #[test]
+    fn module_name_4_parts_dunder_init_py() {
+        assert_debug_snapshot!(non_stdlib_module_name_test_case("foo/bar/baz/__init__.py"), @r###"
+        ModuleName(
+            "foo.bar.baz",
+        )
+        "###);
+    }
+
+    #[test]
+    fn join_1() {
```

Comment:
We should avoid numbering tests. It makes it extremelly difficult to understand what a test is testing and, from it, derive what the expected behavior is.